### PR TITLE
BCDA-4772: Update Synthetic ALR Data

### PR DESCRIPTION
**Pull request by katyasoup**
### Fixes [BCDA-4772](https://jira.cms.gov/browse/BCDA-4772)
The synthetic data generated by the tool in `alr_test.go` was out of sync with the latest updates to [Bill's ALR Implementation Guide](https://github.com/CMSgov/bcda-app/tree/BCDA-4404_alr_mockup). This PR addresses some of those discrepancies; additional work will be needed as further tweaks to the IG are made.

### Change Details
Adds missing fields and updates FHIR data types as needed to make generated synthetic data more closely match the examples from the IG.

### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
Fields identified in this [Jira ticket](https://jira.cms.gov/browse/BCDA-4772) are present in newly generated synthetic data. _(Resolved discrepancies are indicated by a ~strikethrough~; outstanding discrepancies are in **bold**)._

### Feedback Requested
Feel free to pull this branch and [run the command](https://github.com/CMSgov/bcda-app/blob/497ce797193a7a8e77d00e3f1ef5f2d03e5b73d1/bcda/models/fhir/alr/alr_test.go#L29) to generate your own synthetic data to validate against the IG

### CODE REVIEW

File: bcda/models/fhir/alr/v1/coverage.go [hunk 1]
```diff
@@ -15,28 +15,35 @@ func coverage(mbi string, keyValue []utils.KvPair, lastUpdated time.Time) *fhirm
...
+	// 		OrganizationId: &fhirdatatypes.ReferenceId{Value: "example-org-id"},
+	// 	}}
+	// coverage.Payor = append(coverage.Payor, payorRef)
+
 	coverage.Meta = &fhirdatatypes.Meta{
 		LastUpdated: &fhirdatatypes.Instant{
 			Precision: fhirdatatypes.Instant_SECOND,
 			ValueUs:   lastUpdated.UnixNano() / int64(time.Microsecond),
 		},
+		Profile: []*fhirdatatypes.Uri{
```
Comment (CosmicSyntax) [inline bcda/models/fhir/alr/v1/coverage.go:36]: 👍

**Review by CosmicSyntax — APPROVED** [review]
🎉

File: bcda/models/fhir/alr/v1/coverage.go [hunk 2]
```diff
@@ -15,28 +15,35 @@ func coverage(mbi string, keyValue []utils.KvPair, lastUpdated time.Time) *fhirm
...
+			PatientId: &fhirdatatypes.ReferenceId{Value: mbi},
 		},
 	}
+	// coverage.Payor = []*fhirdatatypes.Reference{}
+	// payorRef := &fhirdatatypes.Reference{
+	// 	Reference: &fhirdatatypes.Reference_OrganizationId{
+	// 		OrganizationId: &fhirdatatypes.ReferenceId{Value: "example-org-id"},
+	// 	}}
+	// coverage.Payor = append(coverage.Payor, payorRef)
+
```
Comment (jphechter) [inline bcda/models/fhir/alr/v1/coverage.go:30]: Should we just delete these lines?

File: bcda/models/fhir/alr/v2/coverage.go [hunk 3]
```diff
@@ -18,39 +18,47 @@ func coverage(mbi string, keyValue []utils.KvPair, lastUpdated time.Time) *r4Mod
...
+			PatientId: &r4Datatypes.ReferenceId{Value: mbi},
 		},
 	}
+	// coverage.Payor = []*r4Datatypes.Reference{}
+	// payorRef := &r4Datatypes.Reference{
+	// 	Reference: &r4Datatypes.Reference_OrganizationId{
+	// 		OrganizationId: &r4Datatypes.ReferenceId{Value: "example-org-id"},
+	// 	}}
+	// coverage.Payor = append(coverage.Payor, payorRef)
+
```
Comment (jphechter) [inline bcda/models/fhir/alr/v2/coverage.go:33]: same as above

**Review by jphechter — APPROVED** [review]
LGTM! 👍

File: bcda/models/fhir/alr/v1/coverage.go (hunk 2 above)
Comment (katyasoup) [reply bcda/models/fhir/alr/v1/coverage.go:30]: Ah, waiting on feedback from Bill on these - will adjust accordingly in the next round!

**Pull request merged**